File: BackEnd/BackEnd.AppService/Domain/Registration/Models/Requests/UpdateEmployeeRequest.cs
using BackEnd.Repository.Entities;

namespace BackEnd.AppService.Domain.Registration.Models.Requests;

public record UpdateEmployeeRequest
{
    public Guid Id { get; set; } 
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Password { get; set; }
    public DateTime DateOfBirth { get; set; }
    public string Phone { get; set; }
    public Guid ManagerId { get; set; }
    public Guid PositionEmployeeId { get; set; }
}

File: front/src/app/models/requests/update-employee.request.ts
import { PhoneEmployeeResponse } from "../responses/employee.response";

export class UpdateEmployeeRequest {
  id = '00000000-0000-0000-0000-000000000000';
  firstName: string = '';
  lastName: string = '';
  password: string = '';
  dateOfBirth: Date = new Date();
  dateOfBirth2: string = '';
  phone: string = '';
  email: string = '';
  managerId = '00000000-0000-0000-0000-000000000000';
  managerName = '';
  positionEmployeeId = '00000000-0000-0000-0000-000000000000';
  docNumber = '';
  phones: PhoneEmployeeResponse[] = [];
}
